lib/trees/avl_tree.dart

```diff
@@ -8,31 +7,37 @@ class AvlNode<V extends Comparable> extends BinaryNodeADT<AvlNode<V>, V> {
   /// Difference between height of left and right subtree.
   ///
   /// [balanceFactor] ∈ `{-1, 0, 1}`.
-  /// Any [AvlNode] having [balanceFactor] outside this set is imbalanced.
+  /// Any [AvlNode] having [balanceFactor] that does not belong to this set is
+  /// considered imbalanced.
+  ///
+  /// In [AvlTree] it is the primary focus to actively balance out all
+  /// imbalanced [node]s following addition or deletion.
   int balanceFactor = 0;
```

Comment:
How about using a Map here? Combines the good of both enum and current definition
'rightHeavy' -> -1
'balanced' -> 0
'leftHeavy' -> 1

Reply:
Actually, I like this implementation 

```dart
enum BalanceState { rightHeavy, balanced, leftHeavy }

class Node {
  BalanceState balanceState = BalanceState.balanced;

  int get balanceFactor {
    switch (balanceState) {
      case BalanceState.rightHeavy:
        return -1;
      case BalanceState.balanced:
        return 0;
      case BalanceState.leftHeavy:
        return 1;
    }
  }
}

void main() {
  var node = Node();

  print(node.balanceState);
  print(node.balanceFactor);
}
```

This prints

```
BalanceState.balanced
0
```

Reply:
It just occured to me, if a node is leftHeavy, it's balance factor can be any natural number. Similarly rightHeavy node's balance factor can be any negative integer. So by definition, it cannot be hardcoded, like I did.